flink-yarn/src/test/java/org/apache/flink/yarn/YarnClusterDescriptorTest.java

```diff
@@ -707,6 +751,25 @@ void testShipArchives() throws IOException {
                 YarnConfigOptions.SHIP_ARCHIVES,
                 Arrays.asList(archive1.getAbsolutePath(), archive2.getAbsolutePath()));
         createYarnClusterDescriptor(flinkConfiguration);
+
+        String archive3 = "hdfs:///flink/archive3.zip";
+        final org.apache.hadoop.conf.Configuration hdConf =
+                new org.apache.hadoop.conf.Configuration();
+        hdConf.set(
+                MiniDFSCluster.HDFS_MINIDFS_BASEDIR, temporaryFolder.toAbsolutePath().toString());
+        try (final MiniDFSCluster hdfsCluster = new MiniDFSCluster.Builder(hdConf).build()) {
+            final org.apache.hadoop.fs.Path hdfsRootPath =
+                    new org.apache.hadoop.fs.Path(hdfsCluster.getURI());
+            hdfsCluster.getFileSystem().createNewFile(new org.apache.hadoop.fs.Path(archive3));
+
+            flinkConfiguration.set(
+                    YarnConfigOptions.SHIP_ARCHIVES,
+                    Arrays.asList(archive1.getAbsolutePath(), archive3));
+            final YarnConfiguration yarnConfig = new YarnConfiguration();
+            yarnConfig.set(
+                    CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, hdfsRootPath.toString());
+            createYarnClusterDescriptor(flinkConfiguration, yarnConfig);
```

Comment:
Could you add a assert to check whether the descriptor.getShipFiles() is as expected?